sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuWindowExpression.scala

```diff
@@ -856,7 +857,90 @@ trait GpuAggregateWindowFunction[T <: Aggregation with RollingAggregation[T]]
   def windowAggregation(inputs: Seq[(ColumnVector, Int)]): AggregationOnColumn[T]
 }
 
-case class GpuRowNumber() extends GpuAggregateWindowFunction[RowNumberAggregation] {
+trait BatchedRunningWindowFixer extends AutoCloseable {
+  /**
+   * Save any state needed from the previous output that will be needed to fix up the next batch.
+   * You don't need to worry about the partition columns. Those will be saved separately.
+   * @param finalOutputColumn the output of fixup.
+   */
+  def updateState(finalOutputColumn: GpuColumnVector): Unit
+
+  /**
+   * Fix up `windowedColumnOutput` with any stored state from previous batches.
+   * @param samePartitionMask a boolean mask with no nulls in it. `true` indicates the row
+   *                          is for the same partition by keys that was the last row in the
+   *                          previous batch. `false` indicates it is not. Only values that are for
+   *                          the same partition by keys should be modified.
+   * @param windowedColumnOutput the output of the windowAggregation without anything
+   *                             fixed/modified.
+   * @return a fixed ColumnVector that was with outputs updated for items that were in the same
+   *         group by key as the last row in the previous batch.
+   */
+  def fixUp(
+      samePartitionMask: Either[GpuColumnVector, Boolean],
+      windowedColumnOutput: GpuColumnVector): GpuColumnVector
+}
+
+/**
+ * For many operations a running window (unbounded preceding to current row) can
+ * process the data without dividing the data up into batches that contain all of the data
+ * for a given group by key set. Instead we store a small amount of state from a previous result
```

Comment:
would be nice to expand on "we store a small amount of state". What state are we keeping?

Reply:
That is up to the operation.